tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/PostFundingValidator.java

```diff
@@ -65,9 +70,22 @@ private FundingTypeDTO checkFundingTypeExists(PostFundingDTO pfDto,
   }
 
   private void checkInfoForFundingType(PostFundingDTO pfDto, FundingTypeDTO matchedFundingTypeDto) {
-    // Should not have fundingDetails populdated when allowDetails in FundingType reference is false
+    // Should not have fundingDetails populated when allowDetails in FundingType reference is false
     if (!matchedFundingTypeDto.isAllowDetails() && pfDto.getInfo() != null) {
       pfDto.getMessageList().add(FUNDING_DETAILS_NOT_ALLOWED);
     }
   }
+
+  private void checkAndSetFundingSubType(PostFundingDTO pfDto, Long fundingTypeId) {
+    UUID fundingSubTypeId = pfDto.getFundingSubTypeId();
+    if (fundingSubTypeId != null) {
+      List<FundingSubTypeDto> allCurrentFundingSubTypes =
+          referenceService.findCurrentFundingSubTypesForFundingTypeId(fundingTypeId);
+      FundingSubTypeDto foundFundingSubTypes = allCurrentFundingSubTypes.stream()
+          .filter(dto -> dto.getId().equals(fundingSubTypeId)).findFirst().orElse(null);
+      if (foundFundingSubTypes == null) {
+        pfDto.getMessageList().add(FUNDING_SUB_TYPE_NOT_FOUND);
+      }
```

Comment:
`foundFundingSubTypes` is only used to drive the condition so you could simplify this by using `Stream.anyMatch()` instead of the `filter().findFIrst().orElse()`.